editor.planx.uk/src/@planx/components/Pay/Public/FeeBreakdown/types.ts

```diff
@@ -1,12 +1,20 @@
 export interface FeeBreakdown {
-  applicationFee: number;
-  total: number;
-  reduction: number;
-  vat: number | undefined;
+  amount: {
+    applicationFee: number;
+    total: number;
+    reduction: number;
+    vat: number | undefined;
+  };
+  reductions: string[];
+  exemptions: string[];
 }
 
 export interface PassportFeeFields {
-  "application.fee.calculated": number;
-  "application.fee.payable": number;
-  "application.fee.payable.vat": number;
-}
+  amount: {
+    "application.fee.calculated": number;
+    "application.fee.payable": number;
+    "application.fee.payable.vat": number;
+  },
+  reductions?: Record<string, boolean>
+  exemptions?: Record<string, boolean>
```

Comment:
The change to this structure was pretty much required in order for this to be parsed by Zod.

Initially I spend a fair bit of time trying to get a "raw" passport in here, but as the type of they keys was a list of known strings, and a list of unknown strings (i.e. just `string`) it wasn't possible to parse in a meaningful way.

Here's an invalid / not meaningful TS interface that shows this - 

```ts
interface PassportFeeFields {
  // Known keys
  "application.fee.calculated": number;
  "application.fee.payable": number;
  "application.fee.payable.vat": number;
  // Unknown keys
  [key: string]: string[];
}
```

<img width="781" alt="image" src="https://github.com/user-attachments/assets/a93a6d34-4670-4531-b9a4-6063eb720cac">

As a result, I've added the `preProcessPassport()` function to map a raw passport to this data shape.

Reply:
Another valid and really nice approach here I haven't dug into (but could be worth trying) would be to use template literal types.

I think this is a super interesting approach, but I suspect it will add more complexity not less. I'm also not sure how Zod would handle this, and I still have questions around if `application.fee` should be hardcoded or dynamic. One to explore another time I think!

```ts
interface PassportFeeFields {
  "application.fee.calculated": number;
  "application.fee.payable": number;
  "application.fee.payable.vat": number;
  [key: `application.fee.reductions.${string}`]: string[];
  [key: `application.fee.exemptions.${string}`]: string[];
}
```